snuba/datasets/storages/processors/replaced_groups.py

```diff
@@ -0,0 +1,110 @@
+from typing import Optional
+
+from snuba import environment, settings
+from snuba.clickhouse.processors import QueryProcessor
+from snuba.clickhouse.query import Query
+from snuba.clickhouse.query_dsl.accessors import get_project_ids_in_query
+from snuba.datasets.errors_replacer import ReplacerState, get_projects_query_flags
+from snuba.query.conditions import not_in_condition
+from snuba.query.expressions import Column, FunctionCall, Literal
+from snuba.request.request_settings import RequestSettings
+from snuba.state import get_config
+from snuba.util import is_condition
+from snuba.utils.metrics.backends.wrapper import MetricsWrapper
+
+CONSISTENCY_ENFORCER_PROCESSOR_ENABLED = "consistency_enforcer_processor_enabled"
+
+metrics = MetricsWrapper(environment.metrics, "processors.replaced_groups")
+
+
+class PostReplacementConsistencyEnforcer(QueryProcessor):
+    """
+    This processor tweaks the query to ensure that groups that have been manipulated
+    by a replacer (like after a deletion) are excluded if they need to be.
+
+    There is a period of time between the replacement executing its query and Clickhouse
+    merging the rows to achieve consistency. During this period of time we either
+    have to remvoe those rows manually or to run the query in FINAL mode.
+    """
+
+    def __init__(
+        self, project_column: str, replacer_state_name: Optional[ReplacerState]
+    ) -> None:
+        self.__project_column = project_column
+        # This is used to allow us to keep the replacement state in redis for multiple
+        # replacers on multiple tables. replacer_state_name is part of the redis key.
+        self.__replacer_state_name = replacer_state_name
+
+    def process_query(self, query: Query, request_settings: RequestSettings) -> None:
+        if request_settings.get_turbo():
```

Comment:
most of the logic here is basically cut paste from the project_extension except for the metrics part